src/renderers/shared/ReactPerf.js

```diff
@@ -13,17 +13,38 @@
 
 var ReactDebugTool = require('ReactDebugTool');
 var warning = require('warning');
+var alreadyWarned = false;
 
 function roundFloat(val, base = 2) {
   var n = Math.pow(10, base);
   return Math.floor(val * n) / n;
 }
 
+function warnInProduction() {
+  if (typeof console !== 'undefined' && !alreadyWarned) {
```

Comment:
Nit pick: let's have two ifs, one to exit early if already warned, and another right around the if. This way it's clearer which is guarding what.